src/renderers/webgl/annotations/text.ts

```diff
@@ -0,0 +1,372 @@
+        this.doubleClick = false
+        this.doubleClickTimeout = undefined
+        this.renderer.onAnnotationDoubleClick?.({ type: 'annotationPointer', x, y, clientX: client.x, clientY: client.y, target: this.annotation, ...pointerKeysFromEvent(event.data.originalEvent) })
+      }
+    }
+  }
+
+  private pointerLeave = (event: PIXI.InteractionEvent) => {
+    if (this.renderer.clickedAnnotation !== undefined || this.renderer.hoveredAnnotation !== this || this.renderer.dragging) return
+
+    this.renderer.hoveredAnnotation = undefined
+
+    this.dirty = true
+    this.renderer.dirty = true
+
+    const { x, y } = this.renderer.root.toLocal(event.data.global)
+    const client = clientPositionFromEvent(event.data.originalEvent)
+    this.renderer.onAnnotationPointerLeave?.({ type: 'annotationPointer', x, y, clientX: client.x, clientY: client.y, target: this.annotation, ...pointerKeysFromEvent(event.data.originalEvent) })
+  }
+
+  private resizePointerDown = (_: PIXI.InteractionEvent) => {
+
+    this.resizeClicked = this.triangleGraphic
+    this.renderer.clickedAnnotation = this
+    ;(this.renderer.app.renderer.plugins.interaction as PIXI.InteractionManager).on('pointermove', this.resizePointerMove)
+    this.renderer.zoomInteraction.pause()
+    this.renderer.dragInteraction.pause()
+    this.renderer.decelerateInteraction.pause()
+
+    return
+  }
+
+  private resizePointerMove = (event: PIXI.InteractionEvent) => {
+    if (this.resizeClicked === undefined || this.interaction === 'drag') return
+
+    const { x, y } = this.renderer.root.toLocal(event.data.global)
+    
+    if (!this.renderer.dragging) {
+      this.renderer.dragging = true
+      this.interaction = 'resize'
+    }
+
+    const triangleOrigin = [
+      this.annotation.x + this.annotation.width, this.annotation.y + this.annotation.height
+    ]
+
+    const dx = x - triangleOrigin[0]
+    const dy = y - triangleOrigin[1]
+
+    const newWidth = this.annotation.width + dx
+    const newHeight = this.annotation.height + dy
+
+    this.renderer.onAnnotationResize?.({ type: 'annotationResize', x, y, dx: newWidth < 15 ? 0 : dx, dy: newHeight < 15 ? 0 : dy, target: this.annotation, ...pointerKeysFromEvent(event.data.originalEvent) })
+
+    return
+  }
+
+  private resizePointerUp = (_: PIXI.InteractionEvent) => {
+    if (this.resizeClicked === undefined) return
+
+    this.renderer.clickedAnnotation = undefined
+    this.resizeClicked = undefined
+
+    ;(this.renderer.app.renderer.plugins.interaction as PIXI.InteractionManager).off('pointermove', this.pointerMove)
+    this.renderer.zoomInteraction.resume()
+    this.renderer.dragInteraction.resume()
+    this.renderer.decelerateInteraction.resume()
+
+    if (this.renderer.dragging) {
+      this.renderer.dragging = false
+      this.interaction = undefined
+    }
+
+    return
+  }
+
+  private clearDoubleClick() {
+    this.doubleClickTimeout = undefined
+    this.doubleClick = false
+  }
+
+
+  delete() {
```

Comment:
do we need to destroy the `triangleGraphic`? what about the containers? might be that destroying the containers will destroy everything in them recursively, so just doing that might be sufficient...

Reply:
we do need to destroy this, totally forgot!